tests/components/light/test_init.py

```diff
@@ -18,479 +18,479 @@
+        user_file.write("I,WILL,NOT,WORK,EVER\n")
 
-        assert not setup_component(
-            self.hass, light.DOMAIN, {light.DOMAIN: {CONF_PLATFORM: "test"}}
-        )
+    assert not await async_setup_component(
+        hass, light.DOMAIN, {light.DOMAIN: {CONF_PLATFORM: "test"}}
+    )
 
-    def test_light_profiles(self):
-        """Test light profiles."""
-        platform = getattr(self.hass.components, "test.light")
-        platform.init()
 
-        user_light_file = self.hass.config.path(light.LIGHT_PROFILES_FILE)
+async def test_light_profiles(hass, mock_storage):
+    """Test light profiles."""
+    platform = getattr(hass.components, "test.light")
+    platform.init()
 
-        with open(user_light_file, "w") as user_file:
-            user_file.write("id,x,y,brightness\n")
-            user_file.write("test,.4,.6,100\n")
-            user_file.write("test_off,0,0,0\n")
+    user_light_file = hass.config.path(light.LIGHT_PROFILES_FILE)
 
-        assert setup_component(
-            self.hass, light.DOMAIN, {light.DOMAIN: {CONF_PLATFORM: "test"}}
-        )
-        self.hass.block_till_done()
+    with open(user_light_file, "w") as user_file:
+        user_file.write("id,x,y,brightness\n")
+        user_file.write("test,.4,.6,100\n")
+        user_file.write("test_off,0,0,0\n")
 
-        ent1, _, _ = platform.ENTITIES
+    assert await async_setup_component(
+        hass, light.DOMAIN, {light.DOMAIN: {CONF_PLATFORM: "test"}}
+    )
+    await hass.async_block_till_done()
 
-        common.turn_on(self.hass, ent1.entity_id, profile="test")
+    ent1, _, _ = platform.ENTITIES
 
-        self.hass.block_till_done()
+    await common.async_turn_on(hass, ent1.entity_id, profile="test")
 
-        _, data = ent1.last_call("turn_on")
+    await hass.async_block_till_done()
 
-        assert light.is_on(self.hass, ent1.entity_id)
-        assert {
-            light.ATTR_HS_COLOR: (71.059, 100),
-            light.ATTR_BRIGHTNESS: 100,
-            light.ATTR_TRANSITION: 0,
-        } == data
+    _, data = ent1.last_call("turn_on")
 
-        common.turn_on(self.hass, ent1.entity_id, profile="test_off")
+    assert light.is_on(hass, ent1.entity_id)
+    assert data == {
+        light.ATTR_HS_COLOR: (71.059, 100),
+        light.ATTR_BRIGHTNESS: 100,
+        light.ATTR_TRANSITION: 0,
+    }
 
-        self.hass.block_till_done()
+    await common.async_turn_on(hass, ent1.entity_id, profile="test_off")
 
-        _, data = ent1.last_call("turn_off")
+    await hass.async_block_till_done()
 
-        assert not light.is_on(self.hass, ent1.entity_id)
-        assert {light.ATTR_TRANSITION: 0} == data
+    _, data = ent1.last_call("turn_off")
 
-    def test_light_profiles_with_transition(self):
-        """Test light profiles with transition."""
-        platform = getattr(self.hass.components, "test.light")
-        platform.init()
+    assert not light.is_on(hass, ent1.entity_id)
+    assert data == {light.ATTR_TRANSITION: 0}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)